classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.inc.php

```diff
@@ -0,0 +1,134 @@
+                $submissionFileId = $submissionFileRow->submissionFileId;
+                $submissionFileFileId = $submissionFileRow->fileId;
+                $submissionFilePath = $submissionFileRow->filePath;
+
+                error_log("Removing submission file revisions that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('submission_file_revisions')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                error_log("Removing review round files that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('review_round_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                error_log("Removing review files that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('review_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                // Create a Filesystem object with the appropriate adapter to access the actual files
+                $umask = Config::getVar('files', 'umask', 0022);
+                $adapter = new LocalFilesystemAdapter(
+                    Config::getVar('files', 'files_dir'),
+                    PortableVisibilityConverter::fromArray([
+                        'file' => [
+                            'public' => self::FILE_MODE_MASK & ~$umask,
+                            'private' => self::FILE_MODE_MASK & ~$umask,
+                        ],
+                        'dir' => [
+                            'public' => self::DIRECTORY_MODE_MASK & ~$umask,
+                            'private' => self::DIRECTORY_MODE_MASK & ~$umask,
+                        ]
+                    ]),
+                    LOCK_EX,
+                    LocalFilesystemAdapter::DISALLOW_LINKS
+                );
+
+                $filesystem = new Filesystem($adapter);
+
+                if ($filesystem->has($submissionFilePath)) {
+                    error_log("A submission file was found at ${submissionFilePath}. Trying to delete it...");
+                    try {
+                        $filesystem->delete($submissionFilePath);
+                        error_log("A submission file at ${submissionFilePath} was successfully deleted.");
+                    } catch (FilesystemException | UnableToDeleteFile $exception) {
+                        $exceptionMessage = $exception->getMessage();
+                        error_log("A submission file was found at ${submissionFilePath} but could not be deleted because of: ${exceptionMessage}.");
+                    }
+                } else {
+                    error_log("A submission file was expected but not found at ${submissionFilePath}.");
```

Comment:
I think an `error_log` in this case is unnecessary since we're removing all related content. If it's already gone for some reason, it's not a problem.

Reply:
Agreed. Removed.